apps/src/code-studio/initApp/loadApp.js

```diff
@@ -425,6 +425,18 @@ window.apps = {
       } else {
         callback({});
       }
+    },
+    prepareForRemix: function () {
+      let prepareForRemix = getAppOptions().prepareForRemix;
+      return new Promise((resolve, reject) => {
+        if (prepareForRemix) {
+          prepareForRemix().then(() => {
+            resolve();
+          });
+        } else {
+          resolve();
+        }
```

Comment:
Given `prepareForRemix` should always return a promise, and some ES6, this can be significantly simplified:
```javascript
prepareForRemix: function () {
  const {prepareForRemix} = getAppOptions(); // Destructuring assignment
  if (prepareForRemix) {
    return prepareForRemix(); // Just return this promise - unless you are wanting to hide a resolve value, somehow.
  }
  return Promise.resolve(); // Return an insta-resolved promise.
}
```